pkg/backtest/exchange.go

```diff
@@ -363,20 +365,35 @@ func (e *Exchange) SubscribeMarketData(startTime, endTime time.Time, extraInterv
 }
 
 func (e *Exchange) ConsumeKLine(k types.KLine) {
-	if k.Interval == types.Interval1m {
-		e.currentTime = k.EndTime.Time()
+	matching, ok := e.matchingBook(k.Symbol)
+	if !ok {
+		log.Errorf("matching book of %s is not initialized", k.Symbol)
+		return
+	}
+	if matching.klineCache == nil {
+		matching.klineCache = make(map[types.Interval]types.KLine)
+	}
 
-		matching, ok := e.matchingBook(k.Symbol)
-		if !ok {
-			log.Errorf("matching book of %s is not initialized", k.Symbol)
-			return
+	kline1m, ok := matching.klineCache[k.Interval]
```

Comment:
how to make sure the k.Interval is 1m ?

Reply:
In kline query sql, we ensure that kline with same endtime will be sorted by start time in decreasing order. Therefore for every time frame, 1m kline will be called in priority and trigger klineCache pop.
We could ensure this behaviour unless there's a missing kline1m (and should cause panic to notify developers to backfill the missing data)